enterprise/backend/src/metabase_enterprise/metabot_v3/dummy_tools.clj

```diff
@@ -0,0 +1,75 @@
+(ns metabase-enterprise.metabot-v3.dummy-tools
+  (:require
+   [cheshire.core :as json]
+   [metabase-enterprise.metabot-v3.tools.create-dashboard-subscription]
+   [metabase-enterprise.metabot-v3.tools.query]
+   [metabase-enterprise.metabot-v3.tools.who-is-your-favorite]
+   [metabase.api.common :as api]
+   [toucan2.core :as t2]))
+
+(defn- get-current-user
+  [_ _ context]
+  {:output (if-let [{:keys [id email first_name last_name]}
+                    (or (some-> api/*current-user* deref)
+                        (t2/select-one [:model/User :id :email :first_name :last_name] api/*current-user-id*))]
+             {:user_id id
+              :name (str first_name " " last_name)
+              :email_address email}
+             {:error "current user not found"})
+   :context context})
+
+(defn- get-dashboard-details
+  [_ {:keys [dashboard_id]} context]
+  {:output (or (t2/select-one [:model/Dashboard :id :description :name] dashboard_id)
+               {:error "dashboard not found"})
+   :context context})
+
+(defn- dummy-tool-messages
+  [tool-id arguments content]
+  (let [call-id (random-uuid)]
+    [{:content    nil
+      :role       :assistant,
+      :tool_calls [{:id        call-id
+                    :name      tool-id
+                    :arguments (json/generate-string arguments)}]}
+
+     {:content      (json/generate-string content)
+      :role         :tool,
+      :tool_call_id call-id}]))
+
+(defn- dummy-get-current-user
+  [context]
+  (let [content (:output (get-current-user :get-current-user {} context))]
+    (dummy-tool-messages :get-current-user {} content)))
+
+(defn- dummy-get-dashboard-details
+  [context]
+  (reduce (fn [messages viewed]
+            (if-let [dashboard-id (when (= (:type viewed) :dashboard)
+                                    (:ref viewed))]
+              (let [arguments {:dashboard_id dashboard-id}
+                    content (-> (get-dashboard-details :get-dashboard-details arguments context)
+                                :output)]
+                (into messages (dummy-tool-messages :get-dashboard-details arguments content)))
+              messages))
+          []
+          (:user-is-viewing context)))
```

Comment:
These are dummy tools. The real work is done by the "real" tools. These are responsible for deciding if the tools should be called, as well as transforming the output into tool invocation+response pairs.